subworkflows/local/binning_refinement.nf

```diff
@@ -0,0 +1,64 @@
+/*
+ * Binning with MetaBAT2 and MaxBin2
+ */
+
+include { DASTOOL_FASTATOCONTIG2BIN as DASTOOL_FASTATOCONTIG2BIN_METABAT2 } from '../../modules/nf-core/modules/dastool/fastatocontig2bin/main.nf'
+include { DASTOOL_FASTATOCONTIG2BIN as DASTOOL_FASTATOCONTIG2BIN_MAXBIN2  } from '../../modules/nf-core/modules/dastool/fastatocontig2bin/main.nf'
+include { DASTOOL_DASTOOL                                         } from '../../modules/nf-core/modules/dastool/dastool/main.nf'
+
+workflow BINNING_REFINEMENT {
+    take:
+    contigs        //
+    bins           // channel: [ val(meta), path(bins) ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    ch_contigs_for_dastool = contigs
+                                .map {
+                                    meta, assembly, bams, bais ->
+                                        def meta_new = meta.clone()
+                                        [ meta_new, assembly ]
+                                }
+                                .dump(tag: "binrefine_contigs")
+    bins.dump(tag: "binrefine_bins")
+
+    ch_bins_for_fastatocontig2bin = bins
+                                    .branch {
+                                        metabat2: it[0]['binner'] == 'MetaBAT2'
+                                        maxbin2:  it[0]['binner'] == 'MaxBin2'
+                                    }
+
+    // Run on each bin file separately
+    DASTOOL_FASTATOCONTIG2BIN_METABAT2 ( ch_bins_for_fastatocontig2bin.metabat2, "fa")
+    DASTOOL_FASTATOCONTIG2BIN_MAXBIN2 ( ch_bins_for_fastatocontig2bin.maxbin2, "fasta")
+
+    ch_fastatocontig2bin_for_dastool = Channel.empty()
```

Comment:
is that line needed?

Reply:
I'm future proofing there for if other binners are added, in which case a user may turn off one of METABAT2 or MAXBIN2, so you may not have one of those initialised.